File: 0695-max-area-of-island/0695-max-area-of-island.cpp
class Solution {
public:
    void color(int i,int j,int& maxi,int& maxj,int& trck, vector<vector<int>>& grid){
        if(i < 0 || j < 0 || i == maxi || j == maxj){
            return; 
        }
        if(grid[i][j] == 1){
            grid[i][j] = 2;
            trck++;
            color(i + 1, j, maxi, maxj, trck, grid);
            color(i - 1, j, maxi, maxj, trck, grid);
            color(i, j - 1, maxi, maxj, trck, grid);
            color(i, j + 1, maxi, maxj, trck, grid);
            
        }
    }
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int maxim = 0;
        int ck = 0;
        int im = grid.size();
        int jm = grid[0].size();
        for(int i = 0; i < grid.size(); i++){
            for(int j = 0; j < grid[0].size(); j++){
                if(grid[i][j] == 1){
                    color(i,j,im, jm, ck, grid);
                    maxim = max(maxim, ck);
                    ck = 0;
                }
            }
        }
        return maxim;
        
    }
};

File: 0695-max-area-of-island/0695-max-area-of-island.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        trck = [0]

        def color(i,j):
            if i < 0 or j < 0 or i == len(grid) or j == len(grid[0]):
                return
            if grid[i][j] == 1:
                trck[0] += 1
                grid[i][j] = 2
                color(i + 1, j)
                color(i - 1, j)
                color(i, j + 1)
                color(i, j - 1)

        maxim = 0
        
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:

                    color(i,j)
                    maxim = max(maxim, trck[0])
                    trck[0] = 0
        return maxim